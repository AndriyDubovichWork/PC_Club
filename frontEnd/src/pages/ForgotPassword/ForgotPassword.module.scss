.login {
  width: 80%;
  height: 80vh;
  margin: 0 auto;

  display: flex;

  flex-direction: column;
  align-items: center;
  justify-content: center;
  .form {
    display: flex;
    flex-direction: column;

    align-items: center;
    .field {
      margin: 1vh 0;
    }
  }
}
