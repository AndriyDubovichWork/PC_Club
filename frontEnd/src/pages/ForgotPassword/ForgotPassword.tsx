import React, { useContext } from 'react';
import style from './ForgotPassword.module.scss';
import { Field, Form, Formik } from 'formik';
import LoginSchema from './Validation/ForgotPasswordSchema';
import { TextField } from 'formik-mui';
import Button from '../../Styled/Button/Button';
import { useNavigate } from 'react-router-dom';
import { IsAuthorisedContext } from '../../Context/IsAuthorisedContext';

const ForgotPassword = () => {
  const navigate = useNavigate();

  const initialValues = {
    email: '',
  };
  return (
    <main className={style.login}>
      <Formik
        initialValues={initialValues}
        onSubmit={() => {
          navigate('/PC_Club');
        }}
        validationSchema={LoginSchema}
      >
        <Form className={style.form}>
          <Field
            name='email'
            component={TextField}
            placeholder='Email'
            className={style.field}
          />

          <Button type='submit'>Request Reset Link</Button>
        </Form>
      </Formik>
    </main>
  );
};

export default ForgotPassword;
