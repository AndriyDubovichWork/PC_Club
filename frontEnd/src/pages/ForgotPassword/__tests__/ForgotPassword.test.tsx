import {
  render,
  screen,
  cleanup,
  waitFor,
  fireEvent,
} from '@testing-library/react';
import ForgotPassword from '../ForgotPassword';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('ForgotPassword match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/forgotPassword']}>
        <Routes>
          <Route path='/PC_Club/forgotPassword' element={<ForgotPassword />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('ForgotPassword contains fields', () => {
  const { getByRole } = render(
    <MemoryRouter initialEntries={['/PC_Club/forgotPassword']}>
      <Routes>
        <Route path='/PC_Club/forgotPassword' element={<ForgotPassword />} />
      </Routes>
    </MemoryRouter>
  );

  expect(getByRole('main')).toBeInTheDocument();
  expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
});

test('Register input cnahges', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/forgotPassword']}>
      <Routes>
        <Route path='/PC_Club/forgotPassword' element={<ForgotPassword />} />
      </Routes>
    </MemoryRouter>
  );

  const email: HTMLInputElement = screen.getByPlaceholderText('Email');

  expect(email).toBeInTheDocument();

  waitFor(() => {
    fireEvent.change(email, { target: { value: 'invalid' } });
  });

  expect(email.value).toBe('invalid');
});
