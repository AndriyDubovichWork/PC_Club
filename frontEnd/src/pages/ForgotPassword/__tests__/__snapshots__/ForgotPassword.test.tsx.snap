// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword match snapshot 1`] = `
<main
  className="login"
>
  <form
    action="#"
    className="form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-127p27p-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Request Reset Link
    </button>
  </form>
</main>
`;
